lib/broadway_cloud_pub_sub/producer.ex

```diff
@@ -151,7 +151,7 @@ defmodule BroadwayCloudPubSub.Producer do
         Options.make_token_generator(opts)
       end)
 
-    {specs, opts} = prepare_to_connect(module, client, opts)
+    {specs, opts} = prepare_to_connect(broadway_opts[:name], client, opts)
```

Comment:
this fails a test where the test expects the name of the finch client to be `BroadwayCloudPubSub.ProducerTest.Forwarder.PullClient` but in the test it becomes `PullClient`

Can we expect name to be set (and the test is incorrectly set up), or should it be something like `broadway_opts[:name] || module`?

Reply:
I believe the name is always set, yes. So the test may be missing the remaining broadway options. Is it a unit test that does not go through broadway?

Reply:
yeah, prepare_to_connect tests

I've fixed the tests, although the test setup is a bit clunky cause it pattern matches on keyword lists, I didn't want to be too "intrusive". Also added a test that asserts you can start two topologies